tiledb/sm/fragment/offsets_fragment_metadata.h

```diff
@@ -0,0 +1,736 @@
+/**
+ * @file  offsets_fragment_metadata.h
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2024 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * This file defines class OffsetsFragmentMetadata.
+ */
+
+#ifndef TILEDB_OFFSETS_FRAGMENT_METADATA_H
+#define TILEDB_OFFSETS_FRAGMENT_METADATA_H
+
+#include <deque>
+#include <mutex>
+#include <unordered_map>
+#include <vector>
+
+#include "tiledb/common/common.h"
+#include "tiledb/common/pmr.h"
+#include "tiledb/sm/array_schema/array_schema.h"
+#include "tiledb/sm/filesystem/uri.h"
+#include "tiledb/sm/misc/types.h"
+#include "tiledb/sm/rtree/rtree.h"
+#include "tiledb/sm/storage_manager/context_resources.h"
+
+namespace tiledb {
+namespace sm {
+
+class FragmentMetadata;
+
+/** Collection of lazily loaded fragment metadata */
+class OffsetsFragmentMetadata {
+ public:
+  /* ********************************* */
+  /*          TYPE DEFINITIONS         */
+  /* ********************************* */
+
+  /** Keeps track of which metadata is loaded. */
+  struct LoadedMetadata {
+    bool rtree_ = false;
+    std::vector<bool> tile_offsets_;
+    std::vector<bool> tile_var_offsets_;
+    std::vector<bool> tile_var_sizes_;
+    std::vector<bool> tile_validity_offsets_;
+    std::vector<bool> tile_min_;
+    std::vector<bool> tile_max_;
+    std::vector<bool> tile_sum_;
+    std::vector<bool> tile_null_count_;
+    bool fragment_min_max_sum_null_count_ = false;
+    bool processed_conditions_ = false;
+  };
+
+  /* ********************************* */
+  /*     CONSTRUCTORS & DESTRUCTORS    */
+  /* ********************************* */
+
+  /* Destructor */
```

Comment:
I've noticed a few @param missing in this class. Let's make a quick pass at fixing them.